src/minuteman_mesos_poller.erl

```diff
@@ -286,17 +267,51 @@ extract_bes(FlatVIPBEs) ->
 %% For this reason we have to reverse the ops before applying them
 %% Since the way that it generates this results in this list being reversed
 
-generate_ops(AgentBEIPs, FlatAgentVIPs, LashupVIPs) ->
+generate_ops(AgentIP, AgentVIPs, LashupVIPs) ->
+    FlatAgentVIPs = flatten_vips(AgentVIPs),
     FlatLashupVIPs = flatten_vips(LashupVIPs),
     FlatVIPsToAdd = ordsets:subtract(FlatAgentVIPs, FlatLashupVIPs),
-    FlatLashupVIPsFromThisAgent = lists:filter(
-        fun(#vip_be{backend_ip = BEIP}) -> ordsets:is_element(BEIP, AgentBEIPs) end, FlatLashupVIPs),
+    FlatLashupVIPsFromThisAgent = fetch_vips_from_agent(AgentIP),
     FlatVIPsToDel = ordsets:subtract(FlatLashupVIPsFromThisAgent, FlatAgentVIPs),
+    {AddOps, DelOps} = generate_add_del_ops(FlatVIPsToAdd, FlatVIPsToDel, FlatLashupVIPs),
+    {AddOps2, DelOps2} = generate_add_del_ops2(AgentIP, FlatVIPsToAdd, FlatVIPsToDel, FlatLashupVIPs),
+    {AddOps, DelOps, AddOps2, DelOps2}.
+
+fetch_vips_from_agent(AgentIP) ->
+    LashupVIPs2 = lashup_kv:value(?VIPS_KEY2),
+    FlatLashupVIPs2 = flatten_vips2(LashupVIPs2),
+    FlatLashupVIPsFromThisAgent2 = lists:filter(
```

Comment:
would it be hard to break this `filter` out to a new function and eunit test it with some dummy data?